prboom2/src/g_game.c

```diff
@@ -2004,6 +2004,52 @@ unsigned int GetPackageVersion(void)
   return PACKAGEVERSION;
 }
 
+// [FG] support named complevels on the command line, e.g. "-complevel boom",
+// names based on the args to Chocolate Doom's "-gameversion" parameter
+// and IWAD file names
+
+int G_GetNamedComplevel (const char *arg)
+{
+  int i;
+
+  const struct {
+    int level;
+    const char *const name;
+  } named_complevel[] = {
+    {2, "1.9"},
+    {2, "doom2"},
+    {3, "ultimate"},
+//  {3, "doom"}, // deemed too ambigious
+    {3, "udoom"},
+    {4, "final"},
+    {4, "tnt"},
+    {4, "plutonia"},
+    {9, "boom"},
+    {11, "mbf"},
+  };
+
+  // choose the complevel based on the IWAD
+  if (!strcasecmp(arg, "vanilla"))
+  {
+    if (gamemode == retail)
```

Comment:
If we want to account for Chex, it has the `gamemode` set to `shareware` since it's one episode, but it's based on ultimate doom so the correct complevel is 3 :^)

Reply:
According to Doomwiki, chex.exe is a modified version of Final Doom's executable.

Reply:
Interesting, it's a modified ultimate doom wad but final doom exe 😮

Hm, for consistency with other doom 1 style wads we've always recorded in cl 3. 🤔

Reply:
Whatever, I just checked and current PrBoom+ plays Chex's IWAD demos with Ultimate Doom compatibility.

As far as I know, the only relevant change between the Ultimate and the Final Doom executable was the teleporter z-coordinate bug. However, there was a second Final Doom executable released in 1996, i.e. the same year as Chex, which had this bug fixed. And even Chocolate Doom, i.e. the source port that first introduced specific support for Chex, only checks for the original Final Doom executable to reproduce this bug:

https://github.com/chocolate-doom/chocolate-doom/blob/b26157aca5b12049b35d8dfcf969c51967a369f6/src/doom/p_telept.c#L99

In short terms, I guess it's reasonable and safe to assume Ultimate Doom compatibility for Chex. :wink: